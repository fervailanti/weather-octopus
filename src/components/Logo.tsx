import React, { FC, ReactElement } from 'react'
import { images } from '../images'

export interface LogoProps {
  action: () => void
}

const Logo: FC<LogoProps> = ({ action }): ReactElement => (
  <div className="flex flex-col items-center mt-2 lg:my-4">
    <img
      className="h-16 lg:h-20 transition duration-200 ease-in-out lg:cursor-pointer transform lg:hover:scale-110 active:opacity-50"
      src={images.common.octopus}
      onClick={action}
      alt="octopus"
    />
    <h1 className="text-4xl text-white font-hairline">
      Weather<span className="font-semibold">Octopus</span>
    </h1>
  </div>
)

export default Logo
