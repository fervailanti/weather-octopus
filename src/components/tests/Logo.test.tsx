import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Logo from '../Logo'

describe('Logo Component', () => {
  it('Should render correctly and execute action passed by props', () => {
    const dummyAction = jest.fn()
    const logo = render(<Logo action={dummyAction} />)
    expect(logo.container).toMatchSnapshot()
    expect(logo.container).toHaveTextContent('WeatherOctopus')
    fireEvent.click(screen.getByTestId('octopus-logo'))
    expect(dummyAction).toHaveBeenCalledTimes(1)
  })
})
