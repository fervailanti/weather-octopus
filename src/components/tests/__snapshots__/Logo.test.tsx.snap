// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Component Should render correctly and execute action passed by props 1`] = `
<div>
  <div
    class="flex flex-col items-center mt-2 lg:my-4"
  >
    <img
      alt="octopus"
      class="h-16 lg:h-20 transition duration-200 ease-in-out lg:cursor-pointer transform lg:hover:scale-110 active:opacity-50"
      data-testid="octopus-logo"
      src="octopus.png"
    />
    <h1
      class="text-4xl text-white font-hairline"
    >
      Weather
      <span
        class="font-semibold"
      >
        Octopus
      </span>
    </h1>
  </div>
</div>
`;
